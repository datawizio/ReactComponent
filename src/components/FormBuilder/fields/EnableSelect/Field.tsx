import React, { useCallback } from "react";

import Checkbox from "../../../Checkbox";
import { EnableSelectValueType } from ".";
import { IFormFieldChanged } from "../../types";
import { CheckboxChangeEvent } from "antd/lib/checkbox";

export interface FieldEnableSelectFieldProps {
  name: string | string[];
  placeholder: string;
  value?: EnableSelectValueType;
  renderField: (value: EnableSelectValueType) => any;
  onChange: (change: IFormFieldChanged<EnableSelectValueType>) => void;
}

export const Field: React.FC<FieldEnableSelectFieldProps> = ({
  placeholder,
  value,
  name,
  onChange,
  renderField
}) => {
  const renderValueField = useCallback(() => renderField(value), [
    renderField,
    value.value,
    value.enabled
  ]);

  const handleCheckboxChange = (e: CheckboxChangeEvent) => {
    onChange &&
      onChange({
        name,
        value: {
          value: value.value,
          enabled: e.target.checked
        }
      });
  };

  return (
    <>
      <Checkbox checked={value.enabled} onChange={handleCheckboxChange}>
        {placeholder}
      </Checkbox>
      {renderValueField()}
    </>
  );
};
