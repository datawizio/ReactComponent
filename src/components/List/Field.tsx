import React from "react";

import List from "../List";
import { RightOutlined } from "@ant-design/icons";
import { ListItemFieldProps } from "./types.d";

const ListItemField: React.FC<ListItemFieldProps> = ({
  title,
  description,
  value,
  onClick
}) => {
  return (
    <List.Item onClick={onClick} className="list-item-field">
      <List.Item.Meta title={title} description={description} />
      <div>
        {value} <RightOutlined />
      </div>
    </List.Item>
  );
};

export default ListItemField;
