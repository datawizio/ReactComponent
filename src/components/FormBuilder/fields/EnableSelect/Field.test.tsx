import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import { Field } from "./Field";

const value = {
  enabled: true,
  value: "value"
};
const mockProps = {
  name: "Field name",
  placeholder: "Field placeholder",
  value,
  onChange: jest.fn(),
  renderField: ({ value }) => <span>{value}</span>
};

const setUp = (props?) => mount(<Field {...props} />);

describe("EnableSelect Field component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("Field rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("Field call onChange", () => {
    const mockEvent = {
      target: {
        checked: true
      }
    };
    const checkbox = component.find("Checkbox").first();
    checkbox.props().onChange(mockEvent);
    expect(component.prop("onChange")).toBeCalled();
  });
});
