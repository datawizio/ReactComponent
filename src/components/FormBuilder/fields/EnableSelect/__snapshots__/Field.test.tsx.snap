// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnableSelect Field component Field rendered correctly 1`] = `
<Field
  name="Field name"
  onChange={[MockFunction]}
  placeholder="Field placeholder"
  renderField={[Function]}
  value={
    Object {
      "enabled": true,
      "value": "value",
    }
  }
>
  <Checkbox
    checked={true}
    onChange={[Function]}
  >
    <Checkbox
      checked={true}
      indeterminate={false}
      onChange={[Function]}
    >
      <label
        className="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
      >
        <Checkbox
          checked={true}
          className=""
          defaultChecked={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          prefixCls="ant-checkbox"
          style={Object {}}
          type="checkbox"
        >
          <span
            className="ant-checkbox ant-checkbox-checked"
            style={Object {}}
          >
            <input
              checked={true}
              className="ant-checkbox-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="checkbox"
            />
            <span
              className="ant-checkbox-inner"
            />
          </span>
        </Checkbox>
        <span>
          Field placeholder
        </span>
      </label>
    </Checkbox>
  </Checkbox>
  <span>
    value
  </span>
</Field>
`;
